src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ActiveConfiguredProjectsIgnoringTargetFrameworkProvider.cs

```diff
@@ -0,0 +1,180 @@
+{
+    /// <summary>
+    ///     Provides set of configured projects whose ProjectConfiguration has all dimensions (ignoring the TargetFramework) matching the active VS configuration.
+    ///
+    ///     For example, for a cross-targeting project with TargetFrameworks = "net45;net46" we have:
+    ///     -> All known configurations:
+    ///         Debug | AnyCPU | net45
+    ///         Debug | AnyCPU | net46
+    ///         Release | AnyCPU | net45
+    ///         Release | AnyCPU | net46
+    ///         
+    ///     -> Say, active VS configuration = "Debug | AnyCPU"
+    ///       
+    ///     -> Active configurations ignoring TargetFramework returned by this provider:
+    ///         Debug | AnyCPU | net45
+    ///         Debug | AnyCPU | net46
+    /// </summary>
+    [Export(typeof(ActiveConfiguredProjectsIgnoringTargetFrameworkProvider))]
+    internal class ActiveConfiguredProjectsIgnoringTargetFrameworkProvider : OnceInitializedOnceDisposedAsync
+    {
+        private readonly object _gate = new object();
+        private readonly IUnconfiguredProjectServices _services;
+        private readonly IUnconfiguredProjectCommonServices _commonServices;
+        private readonly IProjectAsynchronousTasksService _tasksService;
+        private readonly IActiveConfiguredProjectSubscriptionService _activeConfiguredProjectSubscriptionService;
+
+        // Cache the last queried TargetFrameworks and associated configured projects per target framework map.
+        // Read/writes for both these fields must be done within a lock to keep them in sync.
+        private ImmutableDictionary<string, ConfiguredProject> _cachedConfiguredProjectsMap;
+        private string _cachedTargetFrameworks;
+
+        private IDisposable _evaluationSubscriptionLink;
+
+        [ImportingConstructor]
+        public ActiveConfiguredProjectsIgnoringTargetFrameworkProvider(
+            IUnconfiguredProjectServices services,
+            IUnconfiguredProjectCommonServices commonServices,
+            [Import(ExportContractNames.Scopes.UnconfiguredProject)]IProjectAsynchronousTasksService tasksService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService)
+            : base(commonServices.ThreadingService.JoinableTaskContext)
+        {
+            Requires.NotNull(services, nameof(services));
+            Requires.NotNull(commonServices, nameof(commonServices));
+            Requires.NotNull(tasksService, nameof(tasksService));
+            Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
+
+            _services = services;
+            _commonServices = commonServices;
+            _tasksService = tasksService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+        }
+
+        [ProjectAutoLoad(startAfter: ProjectLoadCheckpoint.ProjectFactoryCompleted)]
```

Comment:
Why am I only allowed to call this after ProjectFactoryCompleted? Wouldn't it be tied to after the active configuration has been chosen?